employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/common/LocalTimePicker.java

```diff
@@ -0,0 +1,125 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.gwtui.client.common;
+
+import java.time.LocalTime;
+import java.util.Collection;
+import java.util.HashSet;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+
+import com.google.gwt.event.logical.shared.HasValueChangeHandlers;
+import com.google.gwt.event.logical.shared.ValueChangeEvent;
+import com.google.gwt.event.logical.shared.ValueChangeHandler;
+import com.google.gwt.event.shared.GwtEvent;
+import com.google.gwt.event.shared.HandlerRegistration;
+import com.google.gwt.user.client.TakesValue;
+import elemental2.dom.Event;
+import elemental2.dom.HTMLDivElement;
+import elemental2.dom.HTMLInputElement;
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.EventHandler;
+import org.jboss.errai.ui.shared.api.annotations.ForEvent;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+
+@Templated
+public class LocalTimePicker implements TakesValue<LocalTime>,
+                                        HasValueChangeHandlers<LocalTime> {
+
+    @Inject
+    @DataField("date-time-picker-text-field")
+    private HTMLInputElement datePickerTextField;
+
+    @Inject
+    @DataField("date-time-picker")
+    private HTMLDivElement datePickerElement;
+
+    private Collection<ValueChangeHandler<LocalTime>> valueChangeHandlers;
+
+    private BootstrapDateTimePicker datePicker;
+
+    private LocalTime currentValue;
+
+    private static final String DATE_FORMAT_STRING = "LT";
+
+    @PostConstruct
+    public void init() {
+        valueChangeHandlers = new HashSet<>();
+        datePicker = BootstrapDateTimePicker.get(datePickerElement);
+        BootstrapDateTimePicker.BootstrapDateTimePickerOptions options = new BootstrapDateTimePicker.BootstrapDateTimePickerOptions();
+        options.format = DATE_FORMAT_STRING;
+        datePicker.datetimepicker(options);
+        datePicker.on("dp.show", () -> {
+            JQuery.get(datePickerElement).children(".bootstrap-datetimepicker-widget").css("height", "min-content").css("margin-bottom", "-100%")
+                    .css("bottom", "unset");
+        });
+        datePicker.on("dp.change", () -> {
+            String value = datePickerTextField.value;
+            LocalTime oldValue = getValue();
+            if (value != null && !value.isEmpty()) {
+                setValue(LocalTime.parse(MomentJs.moment(value, DATE_FORMAT_STRING).format("HH:mm:ss.SSSSSSSSS")));
+            } else {
+                setValue(null);
+            }
+            ValueChangeEvent.fireIfNotEqual(this, oldValue, getValue());
+        });
+    }
+
+    public boolean reportValidity() {
+        return datePickerTextField.reportValidity();
+    }
+
+    @Override
+    public void setValue(LocalTime value) {
```

Comment:
@Christopher-Chianelli would it be possible to cover also this method by test(s)?

Reply:
Nope; it uses JavaScript native methods, which make it impossible to unit test since it is impossible to mock those methods.